alodataset/coco_detection_dataset.py

```diff
@@ -171,10 +184,12 @@ def getitem(self, idx):
             names=("N", None),
             labels=labels_2d,
         )
-
-        # frame.append_boxes2d(boxes.xcyc().rel_pos())
         frame.append_boxes2d(boxes)
 
+        if self.prepare.return_masks:
+            mask = Panoptic(target["masks"], names=("C", "H", "W"), labels=labels_2d)
```

Comment:
If my understanding is correct the structure is the following:
Frame
    - Boxes 2D
        - label
    - Panoptic
        - label

There is therefore a constraints panoptic.shape[0] must be equal to boxes.shape[0] right ? 
Is there any reason why you choose to not use one mask per boxes as you previously mentioned ?

Let's talk about it during the day

Reply:
Yes. len(labels) == len(Boxes 2D) and there is a constraints with len(labels) == len(Panoptic). So  len(Panoptic) == len(Boxes 2D).

The reason is used Segementation in applications that not required boxes.